registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/RegistrationStatusDao.java

```diff
@@ -256,4 +256,20 @@ public Boolean checkUinAvailabilityForRid(String rid) {
 		return registrationStatusRepositary.createQuerySelect(queryStr, params);
 	}
 
+	public List<RegistrationStatusEntity> getPausedPackets(Integer fetchSize, List<String> statusList) {
+		Map<String, Object> params = new HashMap<>();
+		String className = RegistrationStatusEntity.class.getSimpleName();
+		String alias = RegistrationStatusEntity.class.getName().toLowerCase().substring(0, 1);
+		
+
+		String queryStr = SELECT_DISTINCT + alias + FROM + className + EMPTY_STRING + alias + WHERE + alias

```

Comment:
In this query we have to mainly check if the statusCode = 'PAUSED' and resumeTimestamp > now() and defaultResumeAction is not null, can we remove the latestTransactionStatusCode checks?

Reply:
Also update the unprocess packet query with status not equals to PAUSED check

Reply:
it should be resumeTimestamp < now() as we want packets which has been paused beyond resumetimestamp. There was no PAUSED status code, so I have added it to the enum "RegistrationStatusCode" and implemented it in the query.